src/main/java/com/sap/oss/phosphor/fosstars/tool/MavenScmFinder.java

```diff
@@ -55,7 +58,55 @@
       return Optional.empty();
     }
 
-    return Optional.ofNullable(scm.getUrl());
+    return gitHubUrlParser(scm.getUrl());
+  }
+ 
+  /**
+   * The method tries to resolve a GitHub project path from the given URI syntax.
+   * 
+   * @param url The input URL to be parsed and converted into a GitHub URL.
+   * @return A GitHub URL is parsing is successful. Otherwise an #Optional.empty().
+   * @throws IOException If something goes wrong.
+   */
+  static Optional<String> gitHubUrlParser(String url) throws IOException {
```

Comment:
I probably wouldn't call this method `gitHubUrlParser`. IMHO methods should use verbs and also specify what they actually do. Maybe something like `resolveGitHubProjectPath` or `normalizeGitHubProjectPath` would be better and more understandable here.

Reply:
I agree 👍